net/http/transport_security_state_unittest.cc

```diff
@@ -53,8 +53,9 @@ class TransportSecurityStateTestBase : public ::testing::Test,
                     const std::string& host) {
     SCOPED_TRACE(testing::Message() << network_anonymization_key.ToDebugString()
                                     << " host: " << host);
-    EXPECT_FALSE(state->ShouldUpgradeToSSL(host));
-    EXPECT_FALSE(state->ShouldUpgradeToSSL(network_anonymization_key, host));
+    EXPECT_FALSE(state->ShouldUpgradeToSSL(host, /*is_top_level_nav=*/true));
```

Comment:
shouldn't we also add some tests where this is false?

Reply:
Added in https://github.com/brave/brave-core/pull/27172/commits/b5091e1668729fe37592a355cf0e3889e04c2c6e cc: @goodov to check my assumptions there please.